include/tbbexec/tbb_thread_pool.hpp

```diff
@@ -0,0 +1,413 @@
+#pragma once
+
+#include <tbb/task_arena.h>
+
+#include <exec/static_thread_pool.hpp>
+#include <exec/inline_scheduler.hpp>
+
+
+namespace tbbexec {
+
+template <typename ReceiverId>
+class operation;
+
+using task_base = exec::task_base;
+
+//! This is a P2300-style thread pool wrapping tbb::task_arena, which its docs describe as "A class that represents an
+//! explicit, user-managed task scheduler arena."
+//! Once set up, a tbb::task_arena has
+//! * template<F> void enqueue(F &&f)
+//! and
+//! * template<F> auto execute(F &&f) -> decltype(f())
+//!
+//! See https://spec.oneapi.io/versions/1.0-rev-3/elements/oneTBB/source/task_scheduler/task_arena/task_arena_cls.html
+class tbb_thread_pool {
+    template <typename ReceiverId>
+    friend class operation;
+
+public:
+    tbb_thread_pool() : tbb_thread_pool(tbb::task_arena::attach{}) {}
+
+    explicit tbb_thread_pool(std::uint32_t threadCount) : m_arena{int(threadCount)} { m_arena.initialize(); }
+
+    template <typename... Args>
+    explicit tbb_thread_pool(Args&&... args) : m_arena{std::forward<Args>(args)...} {
+        m_arena.initialize();
+    }
+
+    struct scheduler {
+        using T = scheduler;
+        using ID = scheduler;
+        bool operator==(const scheduler&) const = default;
+
+    private:
+        template <typename ReceiverId>
+        friend class operation;
+
+        class sender {
+        public:
+            using T = sender;
+            using ID = sender;
+            using completion_signatures =
+                stdexec::completion_signatures<stdexec::set_value_t(), stdexec::set_stopped_t()>;
+
+        private:
+            template <typename Receiver>
+            operation<stdexec::__x<std::decay_t<Receiver>>> make_operation_(Receiver&& r) const {
+                return operation<stdexec::__x<std::decay_t<Receiver>>>{this->pool_, (Receiver &&) r};
+            }
+
+            template <class Receiver>
+            friend operation<stdexec::__x<std::decay_t<Receiver>>> tag_invoke(stdexec::connect_t,
+                                                                              sender s,
+                                                                              Receiver&& r) {
+                return s.make_operation_((Receiver &&) r); // std::move(r)?
```

Comment:
Are these moves or forwards? I think they are `std::forward<Receiver>(r)`?

Reply:
Depends on the context :upside_down_face: But in this case it's a forward.

Reply:
Recently I've started taking the receiver by value in the `connect` `tag_invoke` overload. Most receivers are just a pointer to the operation state so copies are cheap, and it simplifies the code.

And yes, in the interest of compile times, I tend to avoid `std::move` and `std::forward`. The fewer templates instantiated, the better.

Reply:
I'm a big fan of
```c++
//! Convenience macro in place of std::forward
//! See https://www.fluentcpp.com/2019/05/14/3-types-of-macros-that-improve-c-code/
//! and https://www.foonathan.net/2020/09/move-forward/
#define FWD(...) static_cast<decltype(__VA_ARGS__)&&>(__VA_ARGS__)
```
which is included basically everywhere in our codebase (next to our `operator""_uz` and `operator""_z` :-) )